compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -404,7 +404,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 let cast = bx.cx().layout_of(self.monomorphize(mir_cast_ty));
 
                 let val = match *kind {
-                    mir::CastKind::PointerExposeAddress => {
+                    mir::CastKind::PointerExpose => {
```

Comment:
FWIW in MiniRust I got rid of this "expose cast", and instead made expose something more like a [NonDivergingIntrinsic](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/enum.NonDivergingIntrinsic.html). That properly reflects that it is entirely decoupled. But ofc that is hard to translate to LLVM...

Reply:
Mmmh, actually... Since currently we generate `ptrtoint` for every conversion, including `addr`, we might be able to add a new intrinsic for exposing that lowers to a noop in LLVM, but can get recognized in miri, and then write `expose` as
```rust
intrinsics::expose(ptr);
ptr.addr()
```

Reply:
> Since currently we generate ptrtoint for every conversion, including addr

Not anymore: https://github.com/rust-lang/rust/pull/121282.

Reply:
But we *could* indeed make it an intrinsic. As a cast it is kind of an odd duck, none of our other casts has a side-effect. All other casts can be removed when their result is unused.